File: Server/src/main/java/bgu/spl171/Packets/ACK.java
package bgu.spl171.Packets;

import bgu.spl171.net.api.bidi.ConnectionsImp;
import bgu.spl171.net.srv.ConnectionHandler;

public class ACK <T> extends Packet{

	final int opCode=4;
	int blockNum;
	
	public ACK(int num) {
		this.blockNum=num;
	}
	public int getOpCode() {
		return opCode;
	}
	public int getblockNum(){
		return blockNum;
	}
	public String toString(){
		return "ACK "+blockNum;
	}
	@SuppressWarnings({ "hiding", "rawtypes", "unchecked" })
	@Override
	public <Packet> void operate(ConnectionsImp c, int connId) {
		((ConnectionHandler)c.getMyHashy().get(connId)).send(new ACK(blockNum));
	}
	
}

File: Client/src/ACK.cpp
//
// Created by david on 17/01/17.
//

#include "../include/Packet.h"
#include "../include/ACK.h"

ACK::ACK(short num):


    blockNum (num)

{}

int ACK::getOpCode()
{
    return opCode;
}
short ACK::getBlockNum()
{
    return blockNum;
}

long ACK::getPacketSize(){
    return -1;
}
std::vector<char> ACK::getData() {
    std::vector<char>* v=new vector<char> {'2','3'};
    return *v;
}
string ACK::getName() {
    return "";
}
void ACK::printError() {
}
string ACK::getErrorMessage() {
    return "";
}


